src/test/java/uk/gov/pay/api/utils/mocks/ChargeResponseFromConnector.java

```diff
@@ -0,0 +1,267 @@
+package uk.gov.pay.api.utils.mocks;
+
+import uk.gov.pay.api.model.CardDetails;
+import uk.gov.pay.api.model.PaymentState;
+import uk.gov.pay.api.model.RefundSummary;
+import uk.gov.pay.api.model.SettlementSummary;
+import uk.gov.pay.commons.model.SupportedLanguage;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+
+public class ChargeResponseFromConnector {
+    private final Long amount, corporateCardSurcharge, totalAmount; 
+    private final PaymentState state; 
+    private final String chargeId, returnUrl, description, reference, email, paymentProvider, gatewayTransactionId, createdDate;
+    private final SupportedLanguage language; 
+    private final boolean delayedCapture; 
+    private final RefundSummary refundSummary; 
+    private final SettlementSummary settlementSummary; 
+    private final CardDetails cardDetails;
+    private final List<Map<?, ?>> links;
+    private final Optional<Map<String, Object>> metadata;
+
+    public Long getAmount() {
+        return amount;
+    }
+
+    public Long getCorporateCardSurcharge() {
+        return corporateCardSurcharge;
+    }
+
+    public Long getTotalAmount() {
+        return totalAmount;
+    }
+
+    public PaymentState getState() {
+        return state;
+    }
+
+    public String getChargeId() {
+        return chargeId;
+    }
+
+    public String getReturnUrl() {
+        return returnUrl;
+    }
+
+    public String getDescription() {
+        return description;
+    }
+
+    public String getReference() {
+        return reference;
+    }
+
+    public String getEmail() {
+        return email;
+    }
+
+    public String getPaymentProvider() {
+        return paymentProvider;
+    }
+
+    public String getGatewayTransactionId() {
+        return gatewayTransactionId;
+    }
+
+    public String getCreatedDate() {
+        return createdDate;
+    }
+
+    public SupportedLanguage getLanguage() {
+        return language;
+    }
+
+    public boolean isDelayedCapture() {
+        return delayedCapture;
+    }
+
+    public RefundSummary getRefundSummary() {
+        return refundSummary;
+    }
+
+    public SettlementSummary getSettlementSummary() {
+        return settlementSummary;
+    }
+
+    public CardDetails getCardDetails() {
+        return cardDetails;
+    }
+
+    public List<Map<?, ?>> getLinks() {
+        return links;
+    }
+
+    public Optional<Map<String, Object>> getMetadata() {
+        return metadata;
+    }
+
+    private ChargeResponseFromConnector(ChargeResponseFromConnectorBuilder builder) {
```

Comment:
I'd be inclined to put the constructor before the getters which might help cut down the impact on readability. Also I'm not sure we discussed `public` fields purely in the context of test code so perhaps worth raising again at some point if you feel it worth a challenge (my only concern is it sets a precedent which someone might miss-interpret and they find their way into `main`).

Reply:
doh. yeah i agree but i don't really want to wait for another long build time. all the more reason to move to kotlin with its implicit getters or relax the use `public final` fields in test classes because it's just code pollution.